packages/syft/src/syft/service/user/user_service.py

```diff
@@ -84,6 +91,207 @@ def create(
         user = result.ok()
         return user.to(UserView)
 
+    @service_method(
+        path="user.forgot_password", name="forgot_password", roles=GUEST_ROLE_LEVEL
+    )
+    def forgot_password(
+        self, context: AuthedServiceContext, email: str
+    ) -> SyftSuccess | SyftError:
+        success_msg = (
+            "If the email is valid, we sent a password "
+            + "reset token to your email or a password request to the admin."
+        )
+        result = self.stash.get_by_email(credentials=context.credentials, email=email)
+        # Isn't a valid email
+        if result.is_err():
+            return SyftSuccess(message=success_msg)
```

Comment:
Should this be a `SyftError`?

Reply:
The goal is to avoid disclosing too much information, which could allow attackers to infer whether someone is a member of the server through spam. So, in this case, we always return a success message, saying that if the email is valid, a message was sent to it.